calcom/cal.com
# Teams billing on creation

## What does this PR do?



This PR adds Stripe billing when creating a new team. Also when deleting a team, the team on Stripe will be deleted.

Fixes #5166 



**Environment**: Staging(main branch) / Production

## Type of change





## How should this be tested?



For both monthly and annual billing
- Create a team
- Invite members and choose the appropriate billing frequency
- In the Stripe checkout session, ensure that the total adds up and the frequency is correct
- Go to the Stripe dashboard to confirm everything is correct
    - Invoice total and frequency
    - Customer information (name & email)
- In the team settings, visit the billing page and the customer portal
    - Visit the user billing page and go to the billing portal. It should be different from the team.
- On the team profile page, delete the team
    - The team should also be deleted on Stripe

## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
